server/models/Dog.js

```diff
@@ -0,0 +1,44 @@
+const mongoose = require("mongoose");
+const currentYear = new Date().getFullYear();
+
+const dogSchema = new mongoose.Schema(
+  {
+    userId: { type: mongoose.Types.ObjectId, required: true, ref: "User" },
+    name: { type: String, required: true, trim: true, maxlength: 50 },
+    breed: { type: String, required: true, trim: true, maxlength: 50 },
+    size: {
+      type: String,
+      required: true,
+      trim: true,
+      lowercase: true,
+      enum: ["xs", "s", "m", "l", "g"],
+    },
+    gender: {
+      type: String,
+      required: true,
+      trim: true,
+      lowercase: true,
+      enum: ["male", "female"],
```

Comment:
```suggestion
      enum: ["male", "female", "other"], or  enum: ["male", "female", "non-binary"],
```
I would recommend adding an additional field here as not all individuals identify as male/female :)